.grit/patterns/go/hidden_goroutine.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Detected a hidden goroutine
+---
+
+Function invocations are expected to synchronous, and this function will execute asynchronously because all it does is call a goroutine. Instead, remove the internal goroutine and call the function using `go`.
+
+tags: #correctness #best-practice
+
+```grit
+language go
+
+file($name, $body) where {
+    $body <: contains `func $func() { go func() { $funcBody }() }` as $hiddenFunc where {
+    } => `func $func() { 
+        $funcBody 
+    }`,
+    $body <: contains `func main(){ $mainBody }` where {
```

Comment:
Add a test case that *doesn't* contain the function in the same file and make sure we still handle that well. What we should be doing:

1. *Always* rewrite function itself.
2. Also rewrite any calls from the same file, if any are found.

Reply:
Hi @morgante, updated above changes